libpod/container_internal.go

```diff
@@ -492,11 +493,16 @@ func (c *Container) setupStorage(ctx context.Context) error {
 		containerInfo, containerInfoErr = c.runtime.storageService.CreateContainerStorage(ctx, c.runtime.imageContext, c.config.RootfsImageName, c.config.RootfsImageID, c.config.Name, c.config.ID, options)
 
 		if !generateName || !errors.Is(containerInfoErr, storage.ErrDuplicateName) {
+			errDuplicateName = true
 			break
 		}
 	}
 	if containerInfoErr != nil {
-		return fmt.Errorf("creating container storage: %w", containerInfoErr)
+		formatedErr := fmt.Errorf("creating container storage: %w", containerInfoErr)
+		if errDuplicateName {
```

Comment:
it would be simpler and more correct to `errors.Is(containerInfoErr, storage.ErrDuplicateName)` here instead of the extra bool.